docs/implementation.md

```diff
@@ -12,9 +12,70 @@
 
 # DID Cache
 
+The DID cache holds most of the state of a Sidetree node. It is a singleton class with the following methods for state update and retrieval.
 
-# Merkle Rooter
+## Record
+
+This is the core method to update state with the following signature.
+
+```javascript
+public record(stateModifyOp: any, blockId: BlockId): URL
```

Comment:
I'd like to change the term from "versionURL" to "operation hash". Because:

1. URL has  a specific meaning in the web world, would be good use another term to disambiguate. 
2. The hash ought to be the same hash as the "hash of an operation used to construct the Merkle tree", so the name corresponds to that explicitly.